### PR TITLE
Freeze time to fix flakey specs

### PR DESCRIPTION
**Pull request by jdelStrother**
Noticed a CI failure here - https://github.com/rack/rack-attack/actions/runs/6932050109/job/18855154765 - which I think is due to the system clock rolling over 15:14:00 while mid-test. How about freezing time in these specs?

### CODE REVIEW

**Comment by jdelStrother** [comment]
Oh I missed that. Will send an update

File: spec/rack_attack_throttle_spec.rb [hunk 1]
```diff
@@ -3,8 +3,9 @@
 require_relative 'spec_helper'
 
 describe 'Rack::Attack.throttle' do
+  freeze_time!
```
Comment (santib) [inline spec/rack_attack_throttle_spec.rb:6]: I tried these changes but it didn't work
image

I think we need to make sure that `Timecop` gets called right before the request is sent (in another before block).

Comment (santib) [reply spec/rack_attack_throttle_spec.rb:6]: also if it's possible to use `Timecop.freeze do` (with block) [I think I'd prefer it](https://github.com/travisjeffery/timecop#timecopsafe_mode) even if it means that we no longer have the requests performed in a before block.

Comment (jdelStrother) [reply spec/rack_attack_throttle_spec.rb:6]: Oh, bah. Minitest::Spec doesn't support defining multiple `before` blocks at the same level 😞

Comment (jdelStrother) [reply spec/rack_attack_throttle_spec.rb]: Thanks for catching that, I didn't test the timecop version well enough. See what you think to the new version.

**Review by santib — APPROVED** [review]

**Comment by santib** [comment]
Thank you @jdelStrother 🙌

**Pull request merged**